compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -272,7 +272,9 @@ object Applications {
     else
       def selectGetter(qual: Tree): Tree =
         val getterDenot = qual.tpe.member(getterName)
-        if (getterDenot.exists) qual.select(TermRef(qual.tpe, getterName, getterDenot))
+        if (getterDenot.exists)
+          getterDenot.accessibleFrom(qual.tpe.widenIfUnstable) // to reset Local
+          qual.select(TermRef(qual.tpe, getterName, getterDenot))
```

Comment:
I think it would be cleaner if `getterDenot.accessibleFrom(...)` was passed into `qual.select` instead of just `getterDenot`.